### PR TITLE
catalan number Via Recurrsion 

### LINKED ISSUE #888: Catalan Number Via Recursion
## 🚀 Name : Catalan Number

Catalan Numbers
Catalan numbers are a sequence of natural numbers that have numerous applications in combinatorics. They are defined based on recursive relationships and have a variety of interpretations in problems involving counting structures like valid sequences of parentheses, binary search trees, paths in grids, and polygon triangulations.

For example:

Balanced Parentheses: The nth Catalan number counts how many valid ways there are to arrange n pairs of parentheses. For n = 3, there are 5 valid sequences: ((())), ()(()), (())(), ()()(), (()()).
Binary Search Trees: It represents the number of distinct binary search trees that can be constructed with n distinct elements.

Recursive Definition:
The recursive formula for Catalan Numbers is:  C(n)= i=0∑n−1 [ C(i)×C(n−i−1) ] , with the base case C(0)=1.
Explanation:
The recursive nature reflects how larger structures can be built from smaller ones. For example, in the case of valid parentheses sequences, placing a pair of parentheses at the outermost positions leaves a smaller valid subsequence inside and another valid subsequence after the outer parentheses.
This recursive division is mirrored in the summation, where C(i) represents the number of valid structures in the left subsequence, and 
C(n−i−1) represents the right subsequence.

Limitations and Considerations Performance: 
For larger values of n, the recursive approach will take significantly longer to compute. This is because of the exponential time complexity and the repeated computation of subproblems.
Optimization: If efficiency is a concern, switching to dynamic programming or using the closed-form formula is recommended for values of n larger than 20. These methods reduce the computational overhead and are more practical for larger inputs.
Stack Depth: Recursive approaches are also limited by the system's recursion depth limit, which may cause a stack overflow for very large values of n (though this is less of a concern for values up to 20)
---

**Would you like to work on this feature?**  
- [✅] Yes

### PR DESCRIPTION
**Pull request by MithanshuHedau**
## 📥 Pull Request

Fixes : #888 
Catalan numbers are a sequence of natural numbers that have numerous applications in combinatorics. They are defined based on recursive relationships and have a variety of interpretations in problems involving counting structures like valid sequences of parentheses, binary search trees, paths in grids, and polygon triangulations.

For example:

Balanced Parentheses: The nth Catalan number counts how many valid ways there are to arrange n pairs of parentheses. For n = 3, there are 5 valid sequences: ((())), ()(()), (())(), ()()(), (()()).
Binary Search Trees: It represents the number of distinct binary search trees that can be constructed with n distinct elements.

Recursive Definition:
The recursive formula for Catalan Numbers is: C(n)= i=0∑n−1 [ C(i)×C(n−i−1) ] , with the base case C(0)=1.
Explanation:
The recursive nature reflects how larger structures can be built from smaller ones. For example, in the case of valid parentheses sequences, placing a pair of parentheses at the outermost positions leaves a smaller valid subsequence inside and another valid subsequence after the outer parentheses.
This recursive division is mirrored in the summation, where C(i) represents the number of valid structures in the left subsequence, and
C(n−i−1) represents the right subsequence.

Limitations and Considerations Performance:
For larger values of n, the recursive approach will take significantly longer to compute. This is because of the exponential time complexity and the repeated computation of subproblems.
Optimization: If efficiency is a concern, switching to dynamic programming or using the closed-form formula is recommended for values of n larger than 20. These methods reduce the computational overhead and are more practical for larger inputs.
Stack Depth: Recursive approaches are also limited by the system's recursion depth limit, which may cause a stack overflow for very large values of n (though this is less of a concern for values up to 20)

### Type of change
- [✅ ] New feature (non-breaking change which adds functionality)
- [✅ ] Documentation update

### Checklist:

- [✅ ] My code follows the style guidelines of this project
- [✅ ] I have performed a self-review of my code
- [✅ ] I have commented my code, particularly in hard-to-understand areas
- [ ✅] I have added tests that prove my fix is effective or that my feature works
- [✅ ] New and existing unit tests pass locally with my changes
- [✅ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by MithanshuHedau** [comment]
@ajay-dhangar , Please Check if All okay or not ,, If so , then can you please merge it my contribution

File: docs/Recursion/catalannumber.md [hunk 1]
```diff
@@ -0,0 +1,102 @@
+---
+id: Catalan Number
```
Comment (ajay-dhangar) [inline docs/Recursion/catalannumber.md]: replace `id: Catalan Number` with `id: catalan-number`

File: docs/Recursion/catalannumber.md [hunk 2]
```diff
@@ -0,0 +1,102 @@
+---
+id: Catalan Number 
+title: Catalan Number Using Recursion
+sidebar_label: Generate Catalan Number 
+description: "The Catalan number sequence is a fundamental concept in combinatorics, appearing in numerous counting problems such as valid parentheses expressions, binary search trees, and polygon triangulations. The nth Catalan number is defined recursively as:
+C(n)= i=0∑n−1
+       C(i)×C(n−i−1) with 𝐶(0)=1 . This recursive structure allows us to break complex problems into smaller subproblems, making Catalan numbers a versatile tool in algorithm design and mathematical theory. This contribution focuses on implementing the recursive approach for generating Catalan numbers."
```
Comment (ajay-dhangar) [inline docs/Recursion/catalannumber.md]: Your description does not follow good practices, and the equation is not allowed. Please carefully read our contribution guidelines.

File: docs/Recursion/catalannumber.md [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+**Binary Search Trees** : It represents the number of distinct binary search trees that can be constructed with `n` distinct elements.
+
+**Recursive Definition:**
+
+The recursive formula for Catalan numbers is:
+
+C(n)= i=0∑n−1
+       C(i)×C(n−i−1)
+
+with the base case C(0)=1.
```
Comment (ajay-dhangar) [inline docs/Recursion/catalannumber.md]: rewrite in a good format based on our syntax and format

File: docs/Recursion/catalannumber.md [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+* The recursive nature reflects how larger structures can be built from smaller ones. For example, in the case of valid parentheses sequences, placing a pair of parentheses at the outermost positions leaves a smaller valid subsequence inside and another valid subsequence after the outer parentheses.
+* This recursive division is mirrored in the summation, where C(i) represents the number of valid structures in the left subsequence, and C(n−i−1) represents the right subsequence.
+
+#### Complexity :
+
+**Dynamic Programming** : A more efficient approach is to use **dynamic programming** to store the results of previous computations in a table and reuse them, reducing the time complexity to O(n^2). This avoids recalculating previously computed values and significantly improves performance.
+
+**Closed-Form Formula** : Another alternative is to use the closed-form formula for Catalan numbers:
+
+C(n)=[1/n+1] (2n )
```
Comment (ajay-dhangar) [inline docs/Recursion/catalannumber.md]: you need to use our math equation feature for this

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

File: docs/Recursion/catalannumber.md (hunk 4 above)
Comment (MithanshuHedau) [reply docs/Recursion/catalannumber.md]: @ajay-dhangar please check now ,, All okay or not

**Comment by MithanshuHedau** [comment]
@ajay-dhangar , please check now ,, All okay or not

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**